### DISCUSSION TITLE
New model contribution following #380

### DISCUSSION THREAD

**Discussion by erogol** [coqui-ai/TTS#443] [General]
Hi @Sadam1195  I just wanted to continue our  #380 talk here. 

Do you have a particular language that you like to attack?

**Comment by Sadam1195** [comment]
Hi, I would like to start with Urdu, Hindi, Sindhi, Arabic, Persian, French, Portuguese, Spanish, Romanian, Catalan, Romansh. I can work on any of them but if it was up to me I would love to begin with Urdu or Sindhi.

**Reply by Sadam1195** [reply]
I am not sure how to begin with Urdu/Sindhi as I can't find any good resources on datasets. I see that both sindhi and urdu are supported by espeak-ng but problem is with dataset creation (forced alignments with aeneas). I want to create a roadmap where anyone can use a guided tutorial and create TTS/SST in his/her own voice.

**Reply by erogol** [reply]
then feel free to start from anywhere you like. Urdu and Sindhi sound good!

**Reply by Sadam1195** [reply]
Do you know any good resources for datasets in urdu/sindhi or do I need to create them from scratch?

**Reply by erogol** [reply]
I don't have a personal experience with any of these languages but maybe you can ask on Gitter channel and see if anyone else knows any better.

**Reply by Sadam1195** [reply]
link for Gitter channel?

**Reply by erogol** [reply]
(@JRMeyer knows a dataset.)

**Reply by Sadam1195** [reply]
Then maybe he can share it and I can start things from there instead having to create one from scratch!

**Reply by erogol** [reply]
https://gitter.im/coqui-ai/TTS?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge

**Reply by Sadam1195** [reply]
Thanks, I will ping a post there shortly. I just have to finish one Spanish TTS project then I am all free to contribute all my hours except when I am sleeping.

**Reply by Sadam1195** [reply]
Also, please fix --`continue_path` --`restore_path` issue on the main when you find the time. I couldn't figure it out and that has been bugging me for a while.

**Reply by erogol** [reply]
what is the issue there?

**Reply by Sadam1195** [reply]
Whenever I use  -`restore_path` it just doesn't procced and gets stuck hours on 

`!python TTS/TTS/bin/train_tacotron.py --restore_path ./Models/LJSpeech/ljspeech-ddc-April-20-2021_12+56PM-0000000/checkpoint_7500.pth.tar --config_path config.json | tee > training.log`

fatal: not a git repository (or any parent up to mount point /content)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /content)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

and when I use --`continue_path` it gets stuck on 

`!python TTS/TTS/bin/train_tacotron.py --continue_path ./Models/LJSpeech/ljspeech-ddc-April-20-2021_12+56PM-0000000 | tee > training.log`

fatal: not a git repository (or any parent up to mount point /content)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

Not sure what's causing that. I am using colab to train model.
link to the config :
https://drive.google.com/file/d/1mm8qY2E0BveSzC03S6czBqf3aKlD7M0M/view?usp=sharing

**Reply by JRMeyer** [reply]
> Hi, I would like to start with Urdu, Hindi, Sindhi, Arabic, Persian, French, Portuguese, Spanish, Romanian, Catalan, Romansh. I can work on any of them but if it was up to me I would love to begin with Urdu or Sindhi.

check out the Coqui list of open speech corpora: https://github.com/coqui-ai/open-speech-corpora/

If you don't see something there, also look in the issues on that repo

**Reply by Sadam1195** [reply]
Hi Josh, urdu and sindhi are not there, I checked before.

**Reply by erogol** [reply]
It doesn't look like a TTS issue but more like an issue in your setup or Colab.

**Reply by Sadam1195** [reply]
Okay, thanks. I will sort it out if it's not TTS issue.

**Reply by astricks** [reply]
@Sadam1195 I am currently training a Hindi model. Not fully trained yet, but I'm close.
I basically created my own datasets, one from open source audio and another which is studio recorded. I can certainly share my experience, what worked for me, and possibly a model in the near future. Let me know if I can help.

**Reply by Sadam1195** [reply]
Nice, I skipped Hindi knowing that someone would work on it as opposed to Urdu/Sindhi. Can you elaborate your process for creating your own dataset? Like how did you made short audios aligned to the sub-text?

**Reply by astricks** [reply]
I tried many techniques - 2 of them worked, one better than the other.

For a lower quality dataset:
1. I downloaded over 25 hours of a particular podcast and wrote a python script to segment it on silences - the result was 11,500 audio clips.
2. I then used google speech-to-text on GCP to transcribe each audio clip. It cost me $300 to get it all transcribed. It isn't 100% perfect, but the model aligns and it actually works.

For a higher quality dataset:
1. I basically created a dataset of thousands of lines and paid someone $2000 to record each line in their home studio. 
2. To create a dataset of thousands of lines, I tried many sources - the only one that really worked for me was downloading subtitles for TV shows and using python scripts to tie them together to get lines of varying length.

Let me know if I can elaborate on any specific step.

**Reply by Sadam1195** [reply]
Hmm, interesting. Though GCP doesn't really works good, it removes any stutters like hmm, umm, eh and also it very robust to noisy speech and often in reality you don't get good quality audios so wouldn't work with the datasets resources I am looking for. 

Second options seems a bit enterprise to me and not many can afford that luxury.

I will share my steps when I create my dataset. 

Thanks for the details.

**Reply by SaadBazaz** [reply]
@Sadam1195 Were you able to make it for Urdu and Sindhi? Did you check out CommonVoice's Urdu dataset? (https://commonvoice.mozilla.org/en/datasets)

**Reply by SaadBazaz** [reply]
@astricks Do you have the Hindi model? Have you shared it online anywhere?

**Reply by Sadam1195** [reply]
> @Sadam1195 Were you able to make it for Urdu and Sindhi? Did you check out CommonVoice's Urdu dataset? (https://commonvoice.mozilla.org/en/datasets)

Hi Saad, I did worked on Urdu and Sindhi models. Unfortunately, I couldn't get permission to make it open source as it was paid work and I had a NDA signed. I trained Tacotron2 , FastSpeech and multiple vocoders to see which worked best. Combination with few of them worked very nicely but as I said I couldn't make it public, I was disheartened to see my efforts go futile as my sole motivation for committing long term to the project was to make it open source as a result I abandoned this community of my future engagements. 
@SaadBazaz 

As far as I know of @astricks he did shared his results and results were decent I'd say but he was working on commercial product from start so he did not made his model public. Maybe he can further clear it up here.
